ush/python/pygw/src/pygw/schema.py

```diff
@@ -0,0 +1,890 @@
+                elif flavor == COMPARABLE:
+                    return_schema["const"] = _to_json_type(s)
+                elif flavor == VALIDATOR and type(s) == Regex:
+                    return_schema["type"] = "string"
+                    return_schema["pattern"] = s.pattern_str
+                else:
+                    if flavor != DICT:
+                        # If not handled, do not check
+                        return return_schema
+
+                    # Schema is a dict
+
+                    required_keys = []
+                    expanded_schema = {}
+                    additional_properties = i
+                    for key in s:
+                        if isinstance(key, Hook):
+                            continue
+
+                        def _key_allows_additional_properties(key):
+                            """Check if a key is broad enough to allow additional properties"""
+                            if isinstance(key, Optional):
+                                return _key_allows_additional_properties(key.schema)
+
+                            return key == str or key == object
+
+                        def _get_key_description(key):
+                            """Get the description associated to a key (as specified in a Literal object). Return None if not a Literal"""
+                            if isinstance(key, Optional):
+                                return _get_key_description(key.schema)
+
+                            if isinstance(key, Literal):
+                                return key.description
+
+                            return None
+
+                        def _get_key_name(key):
+                            """Get the name of a key (as specified in a Literal object). Return the key unchanged if not a Literal"""
+                            if isinstance(key, Optional):
+                                return _get_key_name(key.schema)
+
+                            if isinstance(key, Literal):
+                                return key.schema
+
+                            return key
+
+                        additional_properties = additional_properties or _key_allows_additional_properties(key)
+                        sub_schema = _to_schema(s[key], ignore_extra_keys=i)
+                        key_name = _get_key_name(key)
+
+                        if isinstance(key_name, str):
+                            if not isinstance(key, Optional):
+                                required_keys.append(key_name)
+                            expanded_schema[key_name] = _json_schema(
+                                sub_schema, is_main_schema=False, description=_get_key_description(key)
+                            )
+                            if isinstance(key, Optional) and hasattr(key, "default"):
+                                expanded_schema[key_name]["default"] = _to_json_type(_invoke_with_optional_kwargs(key.default, **kwargs) if callable(key.default) else key.default)
```

Comment:
Ignore pep8 on this line
```suggestion
                                expanded_schema[key_name]["default"] = _to_json_type(_invoke_with_optional_kwargs(key.default, **kwargs) if callable(key.default) else key.default)  # nopep8
```